guest/rust/api_core/src/entity.rs

```diff
@@ -224,20 +225,24 @@ pub fn mutate_component_with_default<T: SupportedValue + Clone + PartialEq>(
 
 /// Adds `child` as a child to `entity`.
 pub fn add_child(entity: EntityId, child: EntityId) {
-    if has_component(entity, children()) {
-        mutate_component(entity, children(), |children| children.push(child));
-    } else {
-        add_component(entity, children(), vec![child]);
+    if has_component(entity, unmanaged_children()) {
+        if has_component(entity, children()) {
+            mutate_component(entity, children(), |children| children.push(child));
+        } else {
+            add_component(entity, children(), vec![child]);
+        }
     }
     add_component(child, parent(), entity);
 }
 
 /// Removes `child` as a child to `entity`.
 pub fn remove_child(entity: EntityId, child: EntityId) {
-    if has_component(entity, children()) {
-        mutate_component(entity, children(), |children| {
-            children.retain(|x| *x != child)
-        });
+    if has_component(entity, unmanaged_children()) {
+        if has_component(entity, children()) {
```

Comment:
this `if` not necessary, `mutate_component` does nothing if it doesn't have that component

Reply:
@philpax Hm yeah, it feels like the mutate_component etc. should return Results, so it's up to the user to handle situations like that? Then it could just be `.ok()` on it to ignore the result.

Reply:
Yeah, that's the plan with #230 / #773, but `mutate_component` already returns an `Option<T>` with the resulting value; you can check for `None` if you want